### ISSUE TITLE
Code needs regenerated to use Quantity object

### ISSUE THREAD

**Issue by lewisheadden** [kubernetes-client/java#209]
I was looking at the `V1ResourceRequirements` object and [noticed these lines](https://github.com/kubernetes-client/java/blob/master/kubernetes/src/main/java/io/kubernetes/client/models/V1ResourceRequirements.java#L36-L40)

```
  @SerializedName("limits")
  private Map<String, String> limits = null;

  @SerializedName("requests")
  private Map<String, String> requests = null;
```

but I think they should be (and when I run `update-client.sh` they are)

```
  @SerializedName("limits")
  private Map<String, Quantity> limits = null;

  @SerializedName("requests")
  private Map<String, Quantity> requests = null;
```

Is there any reason when we regenerated to fix `byte[]` issues we didn't bring this change in too? 

cc/ @brendandburns

**Comment by brendandburns** [comment]
I probably messed something up... I will try to regenerate.

**Comment by brendandburns** [comment]
Actually, I looked into this a little more, this is modelled in swagger as `additionalProperties`

```json
  "io.k8s.api.core.v1.ResourceRequirements": {
    "description": "ResourceRequirements describes the compute resource requirements.",
    "properties": {
     "limits": {
      "description": "Limits describes the maximum amount of compute resources allowed. More info: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/",
      "type": "object",
      "additionalProperties": {
       "$ref": "#/definitions/io.k8s.apimachinery.pkg.api.resource.Quantity"
      }
     },
     "requests": {
      "description": "Requests describes the minimum amount of compute resources required. If Requests is omitted for a container, it defaults to Limits if that is explicitly specified, otherwise to an implementation-defined value. More info: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/",
      "type": "object",
      "additionalProperties": {
       "$ref": "#/definitions/io.k8s.apimachinery.pkg.api.resource.Quantity"
      }
     }
    }
   },
```

And it looks like swagger-codegen and Java doesn't respect `additionalProperties`

https://github.com/swagger-api/swagger-codegen/issues/5187

@mbohlool @lwander fyi

**Comment by lewisheadden** [comment]
@brendandburns I thought this [change to the Maven pom](https://github.com/kubernetes-client/gen/blob/master/openapi/java.xml#L48-L49) changed that behavior?

**Comment by brendandburns** [comment]
I am definitely not an expert here, but I think the problem is that those types never even make it into the Java object model to be replaced.

As far as I can tell, the 'additionalProperties' field in the swagger spec is ignored by the Java code generator.


________________________________________

**Comment by brendandburns** [comment]
I figured this out... There are actually 2 branches in the swagger-codegen project `master` (`2.4.0`) and `3.0.0`

The current gen is using `3.0.0` but the commit for the serialization fix is only in the `master` branch.

It looks like only the `3.0.0` branch supports the additionalProperties stuff.

I'm going to hack up the `gen` project to enable cherry-picks (and hopefully the pick will apply)

**Comment by brendandburns** [comment]
This is fixed with #218
